ios/Handlers/RegistrationHandler.swift

```diff
@@ -76,14 +74,19 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
     }
 
     func cancelCustomRegistration() throws {
-        try handleOTPCode(nil, true)
+        guard let customRegistrationChallenge = self.customRegistrationChallenge else {
+            throw WrapperError.actionNotAllowed(description: "Canceling the custom registration right now is not allowed. Registration is not in progress or pin creation has already started.")
+        }
+        customRegistrationChallenge.sender.cancel(customRegistrationChallenge)
+        handleDidFailToRegister()
     }
 
     func cancelBrowserRegistration() throws {
         guard let browserRegistrationChallenge = self.browserRegistrationChallenge else {
-            throw WrapperError.registrationNotInProgress
+            throw WrapperError.actionNotAllowed(description: "Canceling the browser registration right now is not allowed. Registration is not in progress or pin creation has already started.")
```

Comment:
that description is used twice, can you declare it static and used in both places?

Reply:
They're slightly different, but I can move it to a seperate file where we store strings

Reply:
oh I'm blind (didn't I mention it ;-)